### PR TITLE
#13 Implemented WalletsIn.iterator()

### PR DESCRIPTION
**Pull request by carlosmiranda**
#13: Implemented `WalletsIn.iterator()`. Replaced placeholder unit test with test of actual implementation.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#37](https://github.com/zold-io/java-api/issues/37) is now in scope, role is `REV`

**Comment by 0crat** [comment]
Job [#37](https://github.com/zold-io/java-api/issues/37) is already in scope

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) not enough funds available in the project, can't set budget of job [#37](https://github.com/zold-io/java-api/issues/37), see [§21](http://www.zerocracy.com/policy.html#21); @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) will get no money on completion; in order to fix that, add funds to the project and assign the job again

File: src/main/java/io/zold/api/WalletsIn.java [hunk 1]
```diff
@@ -76,12 +80,23 @@ public Wallet create() {
...
-    // @todo #4:30min Read instance of the Wallet from file and put it
-    //  to the result. Should be taken care of after Wallet interface will have
-    //  necessary implementations. Cover with tests and remove irrelevant test
-    //  case.
     @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public Iterator<Wallet> iterator() {
-        throw new UnsupportedOperationException("iterator() not yet supported");
+        // @checkstyle MagicNumber (1 line)
+        final List<Wallet> wallets = new ArrayList<>(10);
```
Comment (llorllale) [inline src/main/java/io/zold/api/WalletsIn.java]: @carlosmiranda can you use `Directory` and `Filtered` from cactoos here? Also, keep in mind that the default filename extension for wallets is [`.z`](https://github.com/zold-io/zold/pull/282).

**Comment by llorllale** [comment]
@carlosmiranda see above

**Comment by carlosmiranda** [comment]
@llorllale pushed new commits, please check. I added a new file to the test (`6`) without `*.z` extension to demonstrate that this can be filtered as well.

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#37](https://github.com/zold-io/java-api/issues/37) is not in the agenda of @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), can't retrieve data and time of add

**Comment by 0crat** [comment]
Job [#37](https://github.com/zold-io/java-api/issues/37) is not in the agenda of @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), can't retrieve data and time of add

**Comment by 0crat** [comment]
Job [#37](https://github.com/zold-io/java-api/issues/37) is not in the agenda of @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), can't inspect

**Comment by 0crat** [comment]
The job [#37](https://github.com/zold-io/java-api/issues/37) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by ypshenychka** [comment]
@0crat status

**Comment by ypshenychka** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zold-io/java-api/issues/37#issuecomment-428504352))

@ypshenychka This is what I know about this job in [CAZUREFND](https://www.0crat.com/p/CAZUREFND), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#37](https://github.com/zold-io/java-api/issues/37) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 3mon
  * @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) will receive 30 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/CAZUREFND?q=%7Bjob%3A%27gh%3Azold-io%2Fjava-api%2337%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zold-io/java-api/issues/37#issuecomment-429242223))

@ypshenychka This is what I know about this job in [CAZUREFND](https://www.0crat.com/p/CAZUREFND), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#37](https://github.com/zold-io/java-api/issues/37) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 3mon
  * @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) will receive 30 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/CAZUREFND?q=%7Bjob%3A%27gh%3Azold-io%2Fjava-api%2337%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@llorllale According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ypshenychka** [comment]
@llorllale So should I mark it as bad?

**Comment by llorllale** [comment]
@ypshenychka I guess so (I have no idea)

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**